app/settings/data-export/hdx-export.controller.js

```diff
@@ -108,23 +108,67 @@ function (
     }
 
     function selectTag(attribute) {
-        attribute.selected = true;
-        if (attribute.selectedTag && attribute.selectedTag.tag_name) {
-            attribute.pretty = '#' + attribute.selectedTag.tag_name;
-        } else {
-            attribute.pretty = '';
+        attribute.selected = [attribute.id];
+        attribute.hxl_label = createHxlLabel(attribute);
+        if (!attribute.selectedTag || !attribute.selectedTag.tag_name) {
+            addAnother(attribute);
         }
-        attribute.nbAttributes = 1;
-        attribute.selectedHxlAttributes = {};
+
+        attribute.selectedHxlAttributes = [];
     }
 
     function selectHxlAttribute(attribute) {
-        attribute.pretty = '#' + attribute.selectedTag.tag_name;
+        const hxl_attribute = attribute.selectedHxlAttributes[attribute.selectedHxlAttributes.length - 1];
+        if (!needsMatchedAttribute(attribute)) {
+            attribute.hxl_label = createHxlLabel(attribute);
+            return attribute;
+        }
+        // check if we have lat or lon and assign the correct opposite attribute for it
```

Comment:
I have a suggestions that you are free to ignore, drop line 124

and move line 127 - 141 to it's own function called something like addGeoMatchedAttribute()

then at 125 add an else
and put attribute = addGeoMatchedAttribute():
then just return attribute.

If there are other matched attributes later we should consider expanding out to handle them. This is only a suggestion ignore if you hate it.

Reply:
I should say that the naming of the function is actually important so that it can be genrified but I don't know what to call it :/